app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/MapsActivity.java

```diff
@@ -129,33 +148,173 @@ public void onHeaderClick(View view) {
 //        startActivity(goHome);
     }
 
-    private void setUpMenu() {
-        mHomeView.addDrawerListener(
-            new DrawerLayout.DrawerListener() {
-                @Override
-                public void onDrawerSlide(View drawerView, float slideOffset) {
-                    if (slideOffset < 0.4) { mSearchView.setLeftMenuOpen(false); }
+    private void setUpRouteSwitcher() {
+        mRouteIndicator = this.findViewById(R.id.route_indicator);
+        mRouteSwitchLayout = this.findViewById(R.id.route_switcher_layout);
+        mRouteStartInput = this.findViewById(R.id.startloc_input);
+        mRouteEndInput = this.findViewById(R.id.endloc_input);
+        mRouteSwitchButton = this.findViewById(R.id.swap_destination);
+        mRouteSwitchList = this.findViewById(R.id.route_switcher_list);
+
+        mRouteSwitcherAdapter = new RouteSwitcherAdapter(this, new ArrayList<Place>());
+
+        mRouteSwitchList.setAdapter(mRouteSwitcherAdapter);
+        mRouteSwitchList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                switch (focusedView) {
+                    case 1: {
```

Comment:
is there a way to make more descriptive cases, or at least leave comments stating what 1/2 stands for?

Reply:
Added Comment for Clarity